core/dbt/adapters/materialization/models/_materialization.py

```diff
@@ -0,0 +1,143 @@
+    grants: dict = field(default_factory=dict)
+    on_configuration_change: OnConfigurationChangeOption = OnConfigurationChangeOption.default()
+
+    def __str__(self) -> str:
+        """
+        This gets used in some error messages.
+
+        Returns:
+            A user-friendly name to be used in logging, error messages, etc.
+        """
+        return str(self.target_relation)
+
+    def existing_relation(
+        self, describe_relation_results: DescribeRelationResults
+    ) -> Optional[Relation]:
+        """
+        Produce a full-blown `Relation` instance for `self.existing_relation_ref` using metadata from the database
+
+        Args:
+            describe_relation_results: the results from the macro `describe_sql(self.existing_relation_ref)`
+
+        Returns:
+            a `Relation` instance that represents `self.existing_relation_ref` in the database
+        """
+        if self.existing_relation_ref:
+            relation_type = self.existing_relation_ref.type
+            return self.relation_factory.make_from_describe_relation_results(
+                describe_relation_results, relation_type
+            )
+        return None
+
+    @property
+    def intermediate_relation(self) -> Optional[Relation]:
+        if self.target_relation:
+            return self.relation_factory.make_intermediate(self.target_relation)
+        return None
+
+    @property
+    def backup_relation_ref(self) -> Optional[RelationRef]:
+        if self.existing_relation_ref:
+            return self.relation_factory.make_backup_ref(self.existing_relation_ref)
+        return None
+
+    @property
+    def build_strategy(self) -> MaterializationBuildStrategy:
+        return MaterializationBuildStrategy.NoOp
+
+    @property
+    def should_revoke_grants(self) -> bool:
+        """
+        This attempts to mimic the macro `should_revoke()`
+        """
+        should_revoke = {
+            MaterializationBuildStrategy.Alter: True,
+            MaterializationBuildStrategy.Create: False,
+            MaterializationBuildStrategy.NoOp: False,
+            MaterializationBuildStrategy.Replace: True,
+        }
+        return should_revoke[self.build_strategy]
+
+    @classmethod
+    def from_dict(cls, config_dict) -> "Materialization":
+        return cls(**filter_null_values(config_dict))
+
+    @classmethod
+    def from_runtime_config(
+        cls,
+        runtime_config,
+        relation_factory: RelationFactory,
+        existing_relation_ref: Optional[RelationRef] = None,
+    ) -> "Materialization":
+        config_dict = cls.parse_runtime_config(
+            runtime_config, relation_factory, existing_relation_ref
+        )
+        materialization = cls.from_dict(config_dict)
+        return materialization
+
+    @classmethod
+    def parse_runtime_config(
+        cls,
+        runtime_config,
+        relation_factory: RelationFactory,
+        existing_relation_ref: Optional[RelationRef] = None,
+    ) -> dict:
```

Comment:
Not the only place where we're doing this but it's more common in `dbt-core` to to use `typing.Dict` as opposed to `dict` -- it's more flexible and allows you to specify the type of the of keys and values as well

Reply:
Agreed, I use `Dict` where I can. But sometimes the types are unwieldy. I suppose the vast majority fall into `Dict[str, Any]` and I could use that that to slightly more precise than `dict`, which is effectively `Dict[Any, Any]`.